plugins/actions/validate.py

```diff
@@ -121,6 +122,29 @@ def run(self, tmp=None, task_vars=None):
 
         return result
 
+    def expand_all_jinja2_templates(self, variables):
+        for k, v in variables.items():
+            if self._templar.is_template(v):
+                variables[k] = self.expand_jinja2_template(v)
+        return variables
+
+    def expand_jinja2_template(self, var):
+        try:
+            expanded_var = self._templar.template(var, convert_bare=True,
+                                                  fail_on_undefined=True)
+            if expanded_var == var:
+                if not isinstance(expanded_var, string_types):
+                    raise AnsibleUndefinedVariable
+                expanded_var = self._templar.template("{{" + expanded_var + \
+                                                      "}}", convert_bare=True,
+                                                      fail_on_undefined=True)
+        except AnsibleUndefinedVariable as e:
+            expanded_var = u"VARIABLE IS NOT DEFINED!"
+            if self._display.verbosity > 0:
+                expanded_var += u": %s" % str(e)
```

Comment:
same here, this isn't a safe string substitution because of the addition. Unless you are absolutely certain this will never blow up because the values are always ensured to be strings

Reply:
This is a bit confusing, is the `AnsibleUndefinedVariable` exception you are catching here meant to be the one being raised a few lines above? or is this expecting to find that exception coming from other parts? If the former, it might be better to address the problem right in that block, vs. raising the exception to catch it three lines later

Reply:
Yeah, it can be raised by `self._templar.template()` or `self._templar.is_template()`. I am looking at the code for the debug module of Ansible to write this patch.

Umm... now that reflect on this, I think I should catch the exception in `expand_all_jinja2_template()` since it can also be raised by `self._templar.is_template()`.